aptos-move/block-executor/src/view.rs

```diff
@@ -816,59 +866,50 @@ impl<'a, T: Transaction> ResourceState<T> for SequentialState<'a, T> {
-                                //               and logs an error! We need to make this uniform across
-                                //               resources and groups.
-                                *self.incorrect_use.borrow_mut() = true;
-                                error!("Unsync map couldn't patch base value");
-                                return ReadResult::HaltSpeculativeExecution(
-                                    "Unsync map couldn't patch base value".to_string(),
-                                );
-                            },
-                        }
+                        // sequential execution doesn't need to worry about concurrent change going through.
+                        value = exchanged_value;
                     }
                 }
 
-                if let Some(ret) = ReadResult::from_value_with_layout(value, target_kind.clone()) {
-                    if target_kind == ReadKind::Value {
-                        self.read_set
-                            .borrow_mut()
-                            .resource_reads
-                            .insert(key.clone());
-                    }
-
-                    ret
-                } else {
-                    *self.incorrect_use.borrow_mut() = true;
-                    error!(
-                        "Unsync map has RawFromStorage value type, while we are requesting value"
-                    );
-                    ReadResult::HaltSpeculativeExecution(
-                        "Unsync map has RawFromStorage value type, while we are requesting value"
-                            .to_string(),
-                    )
+                match ReadResult::from_value::<T>(value, &target_kind) {
+                    Ok(read_result) => {
+                        // Get read summary in CapturedReads filters only value reads,
+                        // to be consistent, UnsyncReadSet does not record other kinds.
+                        if target_kind == ReadKind::Value {
+                            self.read_set
+                                .borrow_mut()
+                                .resource_reads
+                                .insert(key.clone());
+                        }
+                        Ok(read_result)
+                    },
+                    Err(_) => {
+                        self.read_set.borrow_mut().incorrect_use = true;
+                        error!(
+                            "Unsync map has RawFromStorage value type, while we are requesting value"
+                        );
+                        Ok(ReadResult::HaltSpeculativeExecution(
+                            "Unsync map has RawFromStorage value type, while we are requesting value"
+                                .to_string(),
+                        ))
+                    },
                 }
             },
-            None => ReadResult::Uninitialized,
+            None => Ok(ReadResult::Uninitialized),
```

Comment:
nit: you can lift `Ok` out of this match?

Reply:
done